### PR TITLE
Add tests in TFLite micro for Logistic Int8

### PR DESCRIPTION
**Pull request by giorgio-arenarm**

### CODE REVIEW

File: tensorflow/lite/experimental/micro/kernels/logistic.cc [hunk 1]
```diff
@@ -27,15 +28,92 @@ namespace tflite {
...
+TfLiteStatus CalculateArithmeticOpData(TfLiteContext* context, TfLiteNode* node,
+                                       OpData* data) {
+  const TfLiteTensor* input = GetInput(context, node, kInputTensor);
+  TfLiteTensor* output = GetOutput(context, node, kOutputTensor);
+
+  TF_LITE_ENSURE_EQ(context, input->type, output->type);
+  if (input->type == kTfLiteInt8) {
+    TF_LITE_ENSURE_EQ(context, output->params.zero_point,
+                      std::numeric_limits<int8_t>::min());
+    TF_LITE_ENSURE(context, output->params.scale == 1. / 256);
```
Comment (jenselofsson) [inline tensorflow/lite/experimental/micro/kernels/logistic.cc]: Could TF_LITE_ENSURE_EQ be used here?

Comment (jenselofsson) [inline tensorflow/lite/experimental/micro/kernels/logistic.cc]: Could TF_LITE_ENSURE_EQ be used in this case? The same goes for line 76.

Comment (giorgio-arenarm) [reply tensorflow/lite/experimental/micro/kernels/logistic.cc]: Done

**Comment by rthadur** [comment]
gentle ping @petewarden for review ?

**Review by petewarden — DISMISSED** [review]

**Comment by rthadur** [comment]
@giorgio-arenarm can you please resolve conflicts ?

**Comment by giorgio-arenarm** [comment]
Done, sorry about the delay

**Comment by freddan80** [comment]
@petewarden @advaitjain, gentle ping for review

**Comment by giorgio-arenarm** [comment]
Rebased. @petewarden @advaitjain gentle ping for review

**Review by petewarden — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@giorgio-arenarm Can you please check reviewer comments and keep us posted. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by giorgio-arenarm** [comment]
There is an internal discussion on whether the lookup table implementation is needed for ensuring bit-accurateness. Will update the issue once this is resolved. Sorry for the delay!

**Comment by tensorflowbutler** [comment]
It has been 37 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by giorgio-arenarm** [comment]
Removed LuT implementation and rebased

**Comment by giorgio-arenarm** [comment]
Removed LuT implementation and rebased

**Comment by giorgio-arenarm** [comment]
@petewarden gentle ping for review

**Comment by giorgio-arenarm** [comment]
@petewarden gentle ping for review

**Review by petewarden — DISMISSED** [review]

**Review by rthadur — DISMISSED** [review]

**Comment by rthadur** [comment]
@giorgio-arenarm can you please check this internal errors 
`third_party/tensorflow/lite/micro/kernels/logistic.cc:51:47: error: implicit conversion increases floating-point precision: 'float' to 'double' [-Werror,-Wdouble-promotion]
    TF_LITE_ENSURE_EQ(context, output->params.scale, 1. / 256);
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
./third_party/tensorflow/lite/c/common.h:192:10: note: expanded from macro 'TF_LITE_ENSURE_EQ'
    if ((a) != (b)) {                                                      \
         ^  ~~
third_party/tensorflow/lite/micro/kernels/logistic.cc:51:47: error: implicit conversion increases floating-point precision: 'float' to 'double' [-Werror,-Wdouble-promotion]
    TF_LITE_ENSURE_EQ(context, output->params.scale, 1. / 256);
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
./third_party/tensorflow/lite/c/common.h:194:45: note: expanded from macro 'TF_LITE_ENSURE_EQ'
                         __LINE__, #a, #b, (a), (b));                      \
                         ~~~~~~~~~~~~~~~~~~~^~~~~~~~
./third_party/tensorflow/lite/c/common.h:152:39: note: expanded from macro 'TF_LITE_KERNEL_LOG'
    (context)->ReportError((context), __VA_ARGS__); \
    ~                                 ^~~~~~~~~~~
third_party/tensorflow/lite/micro/kernels/logistic.cc:55:23: error: implicit conversion increases floating-point precision: 'const float' to 'double' [-Werror,-Wdouble-promotion]
        input->params.scale *
        ~~~~~~~~~~~~~~^~~~~ ~
third_party/tensorflow/lite/micro/kernels/logistic.cc:133:62: error: missing field 'profiling_string' initializer [-Werror,-Wmissing-field-initializers]
      activations::LogisticPrepare, activations::LogisticEval};`

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Please sync to master and fix this build error:

```
./tensorflow/lite/micro/kernels/logistic.cc:51:47: error: implicit conversion increases floating-point precision: 'float' to 'double' [-Werror,-Wdouble-promotion]
    TF_LITE_ENSURE_EQ(context, output->params.scale, 1. / 256);
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
./tensorflow/lite/c/common.h:192:10: note: expanded from macro 'TF_LITE_ENSURE_EQ'
    if ((a) != (b)) {                                                      \
         ^  ~~
/tensorflow/lite/micro/kernels/logistic.cc:51:47: error: implicit conversion increases floating-point precision: 'float' to 'double' [-Werror,-Wdouble-promotion]
    TF_LITE_ENSURE_EQ(context, output->params.scale, 1. / 256);
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
./tensorflow/lite/c/common.h:194:45: note: expanded from macro 'TF_LITE_ENSURE_EQ'
                         __LINE__, #a, #b, (a), (b));                      \
                         ~~~~~~~~~~~~~~~~~~~^~~~~~~~
./tensorflow/lite/c/common.h:152:39: note: expanded from macro 'TF_LITE_KERNEL_LOG'
    (context)->ReportError((context), __VA_ARGS__); \
    ~                                 ^~~~~~~~~~~
tensorflow/lite/micro/kernels/logistic.cc:55:23: error: implicit conversion increases floating-point precision: 'const float' to 'double' [-Werror,-Wdouble-promotion]
        input->params.scale *
        ~~~~~~~~~~~~~~^~~~~ ~
tensorflow/lite/micro/kernels/logistic.cc:133:62: error: missing field 'profiling_string' initializer [-Werror,-Wmissing-field-initializers]
      activations::LogisticPrepare, activations::LogisticEval};
                                                             ^
4 errors generated.
```

**Comment by giorgio-arenarm** [comment]
Done rebasing and resolving conflicts

File: tensorflow/lite/micro/kernels/logistic.cc [hunk 2]
```diff
@@ -27,11 +28,59 @@ namespace tflite {
 namespace ops {
 namespace micro {
 namespace activations {
-
+namespace {
```
Comment (mihaimaruseac) [inline tensorflow/lite/micro/kernels/logistic.cc:31]: Why do we need to make this anonymous namespace?

Comment (giorgio-arenarm) [reply tensorflow/lite/micro/kernels/logistic.cc:31]: This utility function is used in this specific file only. I've seen it done in many other reference kernels (see for example depthwise_conv.cc where OpData and CalculateOpData are defined in an anonymous namespace)

Comment (mihaimaruseac) [reply tensorflow/lite/micro/kernels/logistic.cc:31]: Thank you. This makes sense now.

**Review by mihaimaruseac — APPROVED** [review]

**Review by njeffrie — COMMENTED** [review]
Landing this internally with a few changes:
- Remove call to new() from init.  We do not allow for dynamic memory allocation on TFLM
- Moved CalculateArithmeticOpData() to Eval since OpData is now stack allocated in eval
- Removed init, free and prepare methods.
- Changed input_left_shift to int datatype due to compiler error.

In the future, we can transition kernels to use our memory allocation APIs in the place of new(), which means we can cache pre-processed parameters generated in prepare().  Until that update is made, we will see a performance hit.

**Pull request merged**